src/main/resources/META-INF/resources/primefaces/core/core.widget.js

```diff
@@ -622,7 +622,19 @@ if (!PrimeFaces.widget) {
         init: function(cfg) {
             this._super(cfg);
 
-            PrimeFaces.utils.registerDynamicOverlay(this, this.jq, this.id);
+            // do not bind overlay if widget disabled
+            if(cfg.disabled === true) {
+                return;
+            }
+
+            // check if this is a panel based overlay like lightbox or cascadeselect
+            if(this.panel) {
```

Comment:
any nicer way then this?

Reply:
HA I knew you are going to say that let me see if I can do something better.

Reply:
OK I overloaded the constructor of DyanmicWidget to take the two extra params.  If htey are not provided it defaults to `this.jq` and `this.id` like it was already doing before.